includes/class-amp-story-post-type.php

```diff
@@ -2053,4 +2079,107 @@ private static function generate_export( $post_id ) {
 		unlink( $file );
 		die();
 	}
+
+	/**
+	 * Returns the definitions for the stories settings.
+	 *
+	 * @since 1.3
+	 *
+	 * @return array
+	 *
+	 * - meta_args array Arguments passed to `register_meta`; sanitize_callback is required.
+	 * - data      array Any additional data.
+	 */
+	public static function get_stories_settings_meta_definitions() {
+		return apply_filters(
+			'amp_stories_settings_meta_definitions',
+			[
+				'stories_settings_auto_advance_after' => [
+					'meta_args' => [
+						'type'              => 'string',
+						'sanitize_callback' => function( $value ) {
+							$valid_values = [ '', 'auto', 'time', 'media' ];
+
+							if ( ! in_array( $value, $valid_values, true ) ) {
+								return '';
+							}
+							return $value;
+						},
+					],
+					'data'      => [
+						'options' => [
+							[
+								'value'       => '',
+								'label'       => __( 'Manual', 'amp' ),
+								'description' => '',
+							],
+							[
+								'value'       => 'auto',
+								'label'       => __( 'Automatic', 'amp' ),
+								'description' => __( 'Based on the duration of all animated blocks on the page', 'amp' ),
+							],
+							[
+								'value'       => 'time',
+								'label'       => __( 'After a certain time', 'amp' ),
+								'description' => '',
+							],
+							[
+								'value'       => 'media',
+								'label'       => __( 'After media has played', 'amp' ),
+								'description' => __( 'Based on the first media block encountered on the page', 'amp' ),
+							],
+						],
+					],
+				],
+				'stories_settings_auto_advance_after_duration' => [
+					'meta_args' => [
+						'type'              => 'integer',
+						'sanitize_callback' => function( $value ) {
+							$value = intval( $value );
+
+							return filter_var(
+								$value,
+								FILTER_VALIDATE_INT,
+								[
+									'default'   => 0,
+									'min_range' => 1,
+									'max_range' => 100,
+								]
+							);
+						},
+					],
+					'data'      => [],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Adds stories global settings as post meta to all new Stories.
```

Comment:
To clarify: the intention here is that the global setting is applied during post creation, meaning that changing the global setting won't apply to pre-existing stories after the fact?

Reply:
@swissspidy Correct.